lib/jest-matchers/index.ts

```diff
@@ -0,0 +1,20 @@
+import type { ZodTypeAny } from 'zod';
+
+export const toMatchZodSchema = <T extends ZodTypeAny>(
+  val: unknown,
+  schema: T
+) => {
+  const result = schema.safeParse(val);
+
+  if (result.success) {
+    return {
+      pass: true,
+      message: () => 'Value matched schema, but it should not have',
```

Comment:
I'm not sure I understand this message. So it matched but it shouldn't have?

Reply:
This is the error message that's used when the matcher is invoked with `.not`. For example, if someone said `expect(blah).not.toMatchZodSchema(schema)`. Not sure why anyone would do that for this particular matcher, but it's the jest matcher interface:

https://jestjs.io/docs/expect#custom-matchers-api